### PR TITLE
disttask: add handle to run/cancel task and fetch error

### LINKED ISSUE #43065: add handle to submit task

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: close #43065 

Problem Summary:

### What is changed and how it works?
- add handle to submit task 
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/index.go [hunk 1]
```diff
@@ -1809,62 +1808,33 @@ func executeDistGlobalTask(reorgInfo *reorgInfo) error {
...
-
-		if found.State == proto.TaskStateSucceed {
-			return nil
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-ticker.C:
+				if reorgInfo.Job.IsCancelling() {
```
Comment (tangenta) [inline ddl/index.go:1825]: I think the `reorgInfo.Job` is never updated during the execution of global task.

**Review by tangenta — APPROVED** [review]

File: ddl/index.go (hunk 1 above)
Comment (GMHDBJD) [reply ddl/index.go:1865]: Just copying from the original logic.

File: disttask/framework/handle/handle.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+		}
+	}
+
+	ticker := time.NewTicker(checkTaskFinishInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
```
Comment (lance6716) [inline disttask/framework/handle/handle.go:65]: what's the reason of context cancel? Do we need to filter context error or maybe it's better to return nil here?

File: disttask/framework/handle/handle.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+		case <-ticker.C:
+			found, err := globalTaskManager.GetGlobalTaskByID(globalTask.ID)
+			if err != nil {
+				logutil.BgLogger().Info("get global task error", zap.Int64("taskID", globalTask.ID), zap.Error(err))
```
Comment (lance6716) [inline disttask/framework/handle/handle.go:69]: need to prevent printing too many log

File: disttask/framework/handle/handle.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+	"github.com/pingcap/tidb/util/logutil"
+	"go.uber.org/zap"
+)
+
+var (
+	checkTaskFinishInterval = 300 * time.Millisecond
+)
+
+// SubmitGlobalTaskAndRun submits a global task and returns a channel that will be closed when the task is done.
+func SubmitGlobalTaskAndRun(ctx context.Context, taskKey, taskType string, concurrency int, taskMeta []byte) error {
```
Comment (D3Hunter) [inline disttask/framework/handle/handle.go:33]:
```suggestion
func SubmitAndRunGlobalTask(ctx context.Context, taskKey, taskType string, concurrency int, taskMeta []byte) error {
```

File: disttask/framework/handle/handle.go [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+			if err != nil {
+				logutil.BgLogger().Info("get global task error", zap.Int64("taskID", globalTask.ID), zap.Error(err))
+				continue
+			}
+
+			if found == nil {
+				return errors.Errorf("cannot find global task with ID %d", globalTask.ID)
+			}
+
+			if found.State == proto.TaskStateSucceed {
```
Comment (D3Hunter) [inline disttask/framework/handle/handle.go:77]: change to `switch`?

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**